src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorExtensions.cs

```diff
@@ -3522,16 +3522,7 @@ public static T StdDev<T>(in ReadOnlyTensorSpan<T> x)
             TensorPrimitives.Pow((ReadOnlySpan<T>)output, T.CreateChecked(2), output);
```

Comment:
Now knowing that `Sqrt(x)` is not necessarily the same as `Pow(x, 0.5)`; can/should `Pow(x, 2)` be replaces by `x*x`?
I see we use `SumOfSquares` (which uses `x*x`) in `Norm` method.
If this is changed then the `IPowerFunctions` constraint can be dropped.

Reply:
Presumably, all three lines 3521-3523 could be replaced with `SumOfSquares`

Reply:
It is true that `x*x==abs(x)^2` for real numbers. I'm not sure if that necessarily holds for e.g. custom types

Reply:
> It seems that we are using `x*x` instead of `Pow(x, 2)`.

Indeed, what do you suggest?

Reply:
Tagging @tannergooding for thoughts.

Reply:
For IEEE 754 floating-point...

The reason that `Pow(x, 0.5)` isn't necessarily the same as `Sqrt(x)` is due to some of the edge case handling.

For `Sqrt(x)` then any `x >= +0` will produce the nearest representable result, `x == -0` produces `-0`, `x == +∞` produces `+∞`, and `x < 0` will produce `NaN`. The operation is also well understood/constrained so it is significantly faster to handle.

While for `Pow(x, 0.5)` it is generally the same, except that `x == -0` produces `+0` and `x == -∞` produces `+∞`. These differences come about from different consistency requirements/considerations.

-------------------

`Pow(x, 2)` and `x * x` are then semantically equivalent for all inputs and so the latter is always preferred due to it being more efficient.

It may not hold true for custom types, but I think that gets to a point where the benefits to the majority of code out there are significantly better than the potential that some esoteric type violates the consideration and so using `x * x` is reasonable.